parser/src/keys/mod.rs

```diff
@@ -5,19 +5,45 @@ use once_cell::sync::Lazy;
 use parking_lot::Mutex;
 use rustc_hash::FxHashMap as HashMap;
 
-#[cfg(any(target_os = "linux", target_os = "unknown"))]
 mod linux;
```

Comment:
Now both linux and windows OsCode translations are included in builds. If you're not ok with it, a feature flag `parse_every_os_mappings` (or some other name) could be added, which would be normally be disabled, and I would enable it in vscode-kanata

Reply:
`str_to_oscode` is used in parts of the kanata runtime and not just at parse time, for example Linux unicode output and cmd output. So better to keep the original variant executing less code with less branching, and have the vscode-kanata specific function.

Reply:
Is `str_to_oscode` affected by this PR? I haven't touched it, and it doesn't use `OsCode::from_u16` or `OsCode::as_u16`

__________

I've checked all places in `src/kanata` where `OsCode` <-> int conversion occurs:

https://github.com/jtroo/kanata/blob/4bfebe43d9d43ae6281c59cff3811a01021ca30c/src/kanata/mod.rs#L1236

https://github.com/jtroo/kanata/blob/4bfebe43d9d43ae6281c59cff3811a01021ca30c/src/kanata/mod.rs#L991

https://github.com/jtroo/kanata/blob/4bfebe43d9d43ae6281c59cff3811a01021ca30c/src/kanata/mod.rs#L956

https://github.com/jtroo/kanata/blob/4bfebe43d9d43ae6281c59cff3811a01021ca30c/src/kanata/mod.rs#L848

BTW, can't `LAST_PRESSED_KEY` one be stored in `Kanata` as `OsCode` so conversions would be elliminated?

Reply:
Ah right, yes ignore the bit about `str_to_oscode`. Conversions happen in many more places than just what you found. They might be hard to discover due to these trait impls. The overhead should be removed.

```
impl From<u32> for OsCode {
    fn from(item: u32) -> Self {
        Self::from_u16(item as u16).unwrap_or_else(|| panic!("Invalid KeyCode: {item}"))
    }
}

impl From<u16> for OsCode {
    fn from(item: u16) -> Self {
        Self::from_u16(item).unwrap_or_else(|| panic!("Invalid KeyCode: {item}"))
    }
}

impl From<OsCode> for usize {
    fn from(item: OsCode) -> Self {
        item.as_u16() as usize
    }
}

impl From<OsCode> for u32 {
    fn from(item: OsCode) -> Self {
        item.as_u16() as u32
    }
}

impl From<OsCode> for u16 {
    fn from(item: OsCode) -> Self {
        item.as_u16()
    }
}

impl From<&OsCode> for KeyCode {
    fn from(item: &OsCode) -> KeyCode {
        (*item).into()
    }
}
impl From<&KeyCode> for OsCode {
    fn from(item: &KeyCode) -> Self {
        (*item).into()
    }
}
```

Reply:
A note around `LAST_PRESSED_KEY` - yes it should probably be fine to store in the `Kanata` struct. I don't recall why it's not today.

Reply:
> Conversions happen in many more places than just what you found. They might be hard to discover due to these trait impls.

I've found the above 4 conversions by removing impls for From one-by-one and checking locations of errors. These are the only ones I found in `src/kanata`. I haven't listed ones found in parser, since they normally shouldn't matter, unless `src/kanata` uses parser functions [that uses OsCode conversions] after general parsing is done, in fact I haven't checked for that.

> The overhead should be removed.

Not sure what overhead you're talking about here. The trait impls?

Reply:
> So better to keep the original variant executing less code with less branching, and have the vscode-kanata specific function. 

Done exactly that in 32fc783